### DISCUSSION TITLE
Multiple resetting to menu issues.

### DISCUSSION THREAD

**Discussion by fuckgithub42** [sanni/cartreader#385] [Need help]
I recently built the single slot HWv4 with a SNES adapter.
I am able to dump stuff, CRC of ROMs check out even checked the SHA1 on it. SRAM of games work when loaded in an emu, but whenever I want to write something it resets back to the main menu, but it did work once and then never again.
I tried different power supplies, I got one with 2A and 1A. 

I tried to find the code place which seems to be in the writeSRAM function when using 'write save' from the menu.
file ok is printed but dataout is not.
It basically resets back to the menu after I press a button after it shows file ok.

```
println_Msg(F("file ok"));
display_Update();
wait();

// Set pins to output
dataOut();

println_Msg(F("dataOut"));
display_Update();
wait();
```
(I also enabled serial output but I didn't get anything)
 A nintendo power cartridge I tried to write to also had an error saying the blank check failed, but it certainly deleted what was on there.

**Comment by sanni** [comment]
Try with an empty SD card with just the snes.txt file. The file browser can crash if it find a file it doesn't like. The filename of the file you want to write should also be max 8 characters with no special letters or spaces just in case.

**Comment by fuckgithub42** [comment]
I figured it out it seems to be the cart, I tried other games and they work fine.

It is the PAL version of Terranigma with a 64kbit SRAM.
The save on the game is fine though and I can save when playing and also load it just fine.
Since the game uses checksums over the save the SRAM should be fine I guess?

edit:
well I kinda feel stupid now, I cleaned the contacts and now it works, really didn't expect this being able to cause that crash?
edit:
That wasn't actually it, it seems this cart will fail to save write until I did a save read first.
edit:
About the  nintendo power cartridge issue it seems it only deletes the first 2048KB and leave the 2nd 2048KB untouched, can my flash be broken I mean I never used this cart before.
